test/runtests.jl

```diff
@@ -1893,6 +1894,44 @@ end
 
             close(conn)
         end
+
+        @testset "DBInterface integration" begin
+            conn = DBInterface.connect(LibPQ.Connection, "dbname=postgres user=$DATABASE_USER")
+            @test conn isa LibPQ.Connection
+
+            result = DBInterface.execute(
+                conn,
+                "SELECT typname FROM pg_type WHERE oid = 16";
+            )
+            @test result isa LibPQ.Result
+            @test status(result) == LibPQ.libpq_c.PGRES_TUPLES_OK
+            @test isopen(result)
+            @test LibPQ.num_columns(result) == 1
+            @test LibPQ.num_rows(result) == 1
+            @test LibPQ.column_name(result, 1) == "typname"
+            @test LibPQ.column_number(result, "typname") == 1
+            data = columntable(result)
+            @test data[:typname][1] == "bool"
+
+            qstr = "SELECT \$1::double precision as foo, typname FROM pg_type WHERE oid = \$2"
+            stmt = DBInterface.prepare(conn, qstr)
+            result = DBInterface.execute(
+                conn,
+                qstr,
+                (1.0, 16);
+            )
+            @test result isa LibPQ.Result
+            @test status(result) == LibPQ.libpq_c.PGRES_TUPLES_OK
+            @test isopen(result)
+            @test LibPQ.num_columns(result) == 2
+            @test LibPQ.num_rows(result) == 1
+            @test LibPQ.column_name(result, 1) == "foo"
+            @test LibPQ.column_name(result, 2) == "typname"
+
+            DBInterface.close!(result)
+            @test !isopen(result)
```

Comment:
```suggestion
            DBInterface.close!(conn)
            @test !isopen(conn)
```

You could add a dispatch for `Result` as well if you want, but closing the connection is still important

Reply:
Oops that was the original intent, good catch